plugins/arc/src/ArcRenderer/ArcRendering.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { readConfObject } from '@jbrowse/core/configuration'
+import { bpSpanPx, measureText } from '@jbrowse/core/util'
+import { observer } from 'mobx-react'
+import { Tooltip } from 'react-svg-tooltip'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function ArcRendering(props: any) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onClick = (event: any, id: any) => {
+    const { onFeatureClick: handler } = props
+    if (!handler) {
+      return undefined
+    }
+    return handler(event, id)
+  }
+
+  const {
+    features,
+    config,
+    regions,
+    blockKey,
+    bpPerPx,
+    displayModel: { selectedFeatureId },
+  } = props
+  const [region] = regions
+  const arcsRendered = []
+
+  for (const feature of features.values()) {
+    const [left, right] = bpSpanPx(
+      feature.get('start'),
+      feature.get('end'),
+      region,
+      bpPerPx,
+    )
+
+    const featureId = feature.id()
+    const id = blockKey + '-' + featureId
+    let stroke = readConfObject(config, 'color', { feature })
+    let textStroke = 'black'
+    if (
+      selectedFeatureId &&
+      String(selectedFeatureId) === String(feature.id())
+    ) {
+      stroke = textStroke = 'red'
+    }
+    const label = readConfObject(config, 'label', { feature })
+    const caption = readConfObject(config, 'caption', { feature })
+    const strokeWidth = readConfObject(config, 'thickness', { feature })
+    const ref = React.createRef<SVGPathElement>()
+    const tooltipWidth = 20 + measureText(caption?.toString())
+
+    arcsRendered.push(
+      <g key={id} onClick={e => onClick(e, featureId)}>
+        <path
+          id={id}
+          d={`M ${left} 0 C ${left} 100, ${right} 100, ${right} 0`}
+          stroke={stroke}
+          strokeWidth={strokeWidth}
+          fill="transparent"
+          onClick={e => onClick(e, featureId)}
+          ref={ref}
+          pointer-events="stroke"
```

Comment:
I think can use pointerEvents instead of pointer-events syntax to avoid warning from react